experiment/gerrit-onboarder/onboarder.go

```diff
@@ -0,0 +1,357 @@
+			if strings.Contains(line, "inheritFrom") {
+				return strings.TrimSpace(strings.Split(line, "=")[1])
+			}
+		}
+	}
+	return ""
+}
+
+func addSection(header string, configMap map[string][]string, configOrder, neededLines []string) (map[string][]string, []string) {
+	if _, ok := configMap[header]; !ok {
+		configMap[header] = []string{}
+		configOrder = append(configOrder, header)
+	}
+	for _, line := range neededLines {
+		configMap[header] = append(configMap[header], "\t"+line)
+	}
+
+	return configMap, configOrder
+}
+
+func labelExists(configMap map[string][]string) bool {
+	_, ok := configMap[LABEL_HEADER]
+	return ok
+}
+
+func lineInRightHeaderFunc(header, line string) func(map[string][]string) bool {
+	return func(configMap map[string][]string) bool {
+		if value, ok := configMap[header]; ok && contains(value, line) {
+			return true
+		}
+		return false
+	}
+}
+
+func labelAccessExistsFunc(groupName string) func(map[string][]string) bool {
+	return func(configMap map[string][]string) bool {
+		for _, value := range configMap {
+			for _, item := range value {
+				if strings.HasPrefix(strings.TrimSpace(item), "label-Verified =") && strings.HasSuffix(strings.TrimSpace(item), fmt.Sprintf("group %s", groupName)) {
+					return true
+				}
+			}
+		}
+		return false
+	}
+}
+
+func verifyInTree(host, cur_branch string, configMap map[string][]string, verify func(map[string][]string) bool) (bool, error) {
+	if verify(configMap) {
+		return true, nil
+	} else if inheritance := getInheritedRepo(configMap); inheritance != "" {
+		parent_branch := cur_branch + "_parent"
+		if err := fetchMetaConfig(host, inheritance, parent_branch); err != nil {
+			return false, fmt.Errorf("unable to fetch refs/meta/config for %s: %w", inheritance, err)
+		}
+		data, err := ioutil.ReadFile(PROJECT_CONFIG_FILE)
+		if err != nil {
+			return false, fmt.Errorf("faield to read project.config file: %w", err)
+		}
+		newConfig, _ := configToMap(string(data))
+		ret, err := verifyInTree(host, parent_branch, newConfig, verify)
+		if err != nil {
+			return false, fmt.Errorf("faield to check if lines in config for %s/%s: %w", host, inheritance, err)
+		}
+		if err := bumper.Call(os.Stdout, os.Stderr, "git", "checkout", cur_branch); err != nil {
+			return false, fmt.Errorf("failed to checkout %s, %w", cur_branch, err)
+		}
+		if err := bumper.Call(os.Stdout, os.Stderr, "git", "branch", "-D", parent_branch); err != nil {
+			return false, fmt.Errorf("failed to delete %s branch, %w", parent_branch, err)
+		}
+		return ret, nil
+	}
+	return false, nil
+}
+
+func ensureProjectConfig(config, host, cur_branch, groupName string) (string, error) {
+	configMap, orderedKeys := configToMap(config)
+
+	// Check that prow automation robot has access to refs/*
+	accessLines := []string{}
+	readAccessLine := fmt.Sprintf(PROW_READ_ACCESS_FORMAT, groupName)
+	prowReadAccess, err := verifyInTree(host, cur_branch, configMap, lineInRightHeaderFunc(ACCESS_HEADER, readAccessLine))
```

Comment:
Other than `[access "refs/*"]`, it could be anything else like `[access "refs/heads/*"]`, or `[access "refs/heads/main"]`, or even more. Would suggest make it more flexible, like anything matching `[access "refs/.*`